### PR TITLE
 Obg 752: Create tests for "TPP-Redirect-Preferred" header

### PR DESCRIPTION
**Pull request by ivaniuminov**
https://github.com/adorsys/open-banking-gateway/issues/752

### CODE REVIEW

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java [hunk 1]
```diff
@@ -71,6 +72,21 @@ public SELF fintech_calls_list_accounts_for_anton_brueckner(String bankId) {
         return self();
     }
 
+    public SELF fintech_calls_list_accounts_for_anton_brueckner_tpp_redirect_preferred() {
+        ExtractableResponse<Response> response = withAccountsHeaders(ANTON_BRUECKNER, SANDBOX_BANK_ID , requestSigningService, OperationType.AIS)
+                    .header(SERVICE_SESSION_ID, UUID.randomUUID().toString())
+                    .header(TPP_REDIRECT_PREFERRED,"true")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java:78]: Can we define constant for `true` ?

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java [hunk 2]
```diff
@@ -137,6 +153,22 @@ public SELF fintech_calls_list_transactions_for_anton_brueckner() {
         return self();
     }
 
+    public SELF fintech_calls_list_transactions_for_anton_brueckner_tpp_redirect_preferred() {
+        ExtractableResponse<Response> response = withTransactionsHeaders(ANTON_BRUECKNER, requestSigningService, OperationType.AIS, GetTransactionsQueryParams.newEmptyInstance())
+                .header(SERVICE_SESSION_ID, UUID.randomUUID().toString())
+                .header(TPP_REDIRECT_PREFERRED, "true")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java:159]: Can we define constant for `true` ?

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentRequestCommon.java [hunk 3]
```diff
@@ -66,6 +67,42 @@ public SELF fintech_calls_initiate_payment_for_anton_brueckner() {
         return self();
     }
 
+    public SELF fintech_calls_initiate_payment_for_anton_brueckner_tpp_redirect_preferred() {
+        String body = readResource("restrecord/tpp-ui-input/params/anton-brueckner-single-sepa-payment.json");
+        ExtractableResponse<Response> response = withPaymentHeaders(ANTON_BRUECKNER, requestSigningService, PIS, body)
+                .header(TPP_REDIRECT_PREFERRED, "true")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentRequestCommon.java:73]: Here and elsewhere. Can we define the constant for `true` ?

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentRequestCommon.java [hunk 4]
```diff
@@ -66,6 +67,42 @@ public SELF fintech_calls_initiate_payment_for_anton_brueckner() {
...
+        updateServiceSessionId(response);
+        updateRedirectCode(response);
+        updateNextPaymentAuthorizationUrl(response);
+        return self();
+    }
+
+    public SELF fintech_calls_initiate_payment_for_anton_brueckner_tpp_redirect_not_preferred() {
+        String body = readResource("restrecord/tpp-ui-input/params/anton-brueckner-single-sepa-payment.json");
+        ExtractableResponse<Response> response = withPaymentHeaders(ANTON_BRUECKNER, requestSigningService, PIS, body)
+                .header(TPP_REDIRECT_PREFERRED, "false")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentRequestCommon.java:91]: Here and elsewhere. Can we define the constant for `false` ?

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/HeaderNames.java [hunk 5]
```diff
@@ -16,6 +16,7 @@
     public static final String FINTECH_USER_ID = "Fintech-User-ID";
     public static final String SERVICE_SESSION_PASSWORD = "Service-Session-Password";
     public static final String X_XSRF_TOKEN = "X-XSRF-TOKEN";
+    public static final String TPP_REDIRECT_PREFERRED = "TPP-Redirect-Preferred";
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/HeaderNames.java:19]: The tests won't work. We don't allow passing this through API. See 
```
@Service
public class TppRedirectPreferredResolver {

    public Boolean isRedirectApproachPreferred(CurrentBankProfile config) {
        return config.isTryToUsePreferredApproach()
                       ? config.getPreferredApproach() == Approach.REDIRECT
                       : null;
    }
}
```
You need to alter bank configuration to enable this header

Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/HeaderNames.java:19]: Was considered, tests changed

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java (hunk 1 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java:78]: Header addition was removed

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java (hunk 2 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/AccountInformationRequestCommon.java:159]: Header addition was removed

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentRequestCommon.java (hunk 3 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentRequestCommon.java:73]: Header addition was removed

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentRequestCommon.java (hunk 4 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-tests-common/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/stages/PaymentRequestCommon.java:91]: Header addition was removed

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java [hunk 6]
```diff
@@ -28,6 +29,8 @@
  * Happy-path test that uses wiremock-stubbed request-responses to drive banking-protocol.
  */
 @SuppressWarnings("CPD-START") // Makes no sense to be too abstract
+@Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = false, preferred_approach = 'REDIRECT' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java:32]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java [hunk 7]
```diff
@@ -75,6 +78,58 @@ void testPaymentInitializationUsingRedirect(Approach expectedApproach) {
                 .fintech_calls_payment_status();
     }
 
+    @Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = true, preferred_approach = 'REDIRECT' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java:81]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java [hunk 8]
```diff
@@ -105,5 +160,69 @@ void testPaymentInitializationUsingRedirectWithCookieValidation(Approach expecte
...
+                .open_banking_redirect_from_aspsp_ok_webhook_called_for_api_test_without_cookie_unauthorized()
+                .and()
+                .open_banking_redirect_from_aspsp_ok_webhook_called_for_api_test();
+        then()
+                .open_banking_has_stored_payment()
+                .fintech_calls_payment_activation_for_current_authorization_id()
+                .fintech_calls_payment_status();
+    }
+
+    @Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = true, preferred_approach = 'EMBEDDED' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java:196]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java [hunk 9]
```diff
@@ -28,6 +29,8 @@
  * Happy-path test that uses wiremock-stubbed request-responses to drive banking-protocol.
  */
 @SuppressWarnings("CPD-START") // Makes no sense to be too abstract
+@Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = false, preferred_approach = 'REDIRECT' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java:32]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java [hunk 10]
```diff
@@ -91,6 +94,58 @@ void testPaymentInitializationUsingRedirect(Approach expectedApproach) {
                 .fintech_calls_payment_status();
     }
 
+    @Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = true, preferred_approach = 'REDIRECT' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java:97]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java [hunk 11]
```diff
@@ -91,6 +94,58 @@ void testPaymentInitializationUsingRedirect(Approach expectedApproach) {
...
+                .user_anton_brueckner_sees_that_he_needs_to_be_redirected_to_aspsp_and_redirects_to_aspsp()
+                .and()
+                .open_banking_redirect_from_aspsp_ok_webhook_called_for_api_test();
+        then()
+                .open_banking_has_stored_payment()
+                .fintech_calls_payment_activation_for_current_authorization_id()
+                .fintech_calls_payment_status();
+    }
+
+    @Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = true, preferred_approach = 'EMBEDDED' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java:123]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java [hunk 12]
```diff
@@ -121,5 +176,69 @@ void testPaymentInitializationUsingRedirectWithCookieValidation(Approach expecte
                 .fintech_calls_payment_activation_for_current_authorization_id()
                 .fintech_calls_payment_status();
     }
+
+    @Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = true, preferred_approach = 'REDIRECT' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java:180]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java [hunk 13]
```diff
@@ -121,5 +176,69 @@ void testPaymentInitializationUsingRedirectWithCookieValidation(Approach expecte
...
+                .open_banking_redirect_from_aspsp_ok_webhook_called_for_api_test_without_cookie_unauthorized()
+                .and()
+                .open_banking_redirect_from_aspsp_ok_webhook_called_for_api_test();
+        then()
+                .open_banking_has_stored_payment()
+                .fintech_calls_payment_activation_for_current_authorization_id()
+                .fintech_calls_payment_status();
+    }
+
+    @Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = true, preferred_approach = 'EMBEDDED' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java:212]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java [hunk 14]
```diff
@@ -39,6 +39,7 @@
 (and it doesn't make sense to import sandbox module here as it is XS2A test) moving it back to plain xs2a bean:
  */
 @Sql(statements = "UPDATE opb_bank_action SET protocol_bean_name = 'xs2aListTransactions' WHERE protocol_bean_name = 'xs2aSandboxListTransactions'")
+@Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = false, preferred_approach = 'REDIRECT' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java:42]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java [hunk 15]
```diff
@@ -87,6 +88,58 @@ void testAccountsListWithConsentUsingRedirect(Approach expectedApproach) {
...
+                .user_anton_brueckner_sees_that_he_needs_to_be_redirected_to_aspsp_and_redirects_to_aspsp()
+                .and()
+                .open_banking_redirect_from_aspsp_ok_webhook_called_for_api_test();
+        then()
+                .open_banking_has_consent_for_anton_brueckner_account_list()
+                .fintech_calls_consent_activation_for_current_authorization_id()
+                .open_banking_can_read_anton_brueckner_account_data_using_consent_bound_to_service_session();
+    }
+
+    @Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = true, preferred_approach = 'EMBEDDED' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java:117]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java [hunk 16]
```diff
@@ -114,6 +167,62 @@ void testTransactionsListWithConsentUsingRedirect(Approach expectedApproach) {
                 );
     }
 
+    @Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = true, preferred_approach = 'REDIRECT' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java:170]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java [hunk 17]
```diff
@@ -114,6 +167,62 @@ void testTransactionsListWithConsentUsingRedirect(Approach expectedApproach) {
...
+                .open_banking_redirect_from_aspsp_ok_webhook_called_for_api_test();
+        then()
+                .open_banking_has_consent_for_anton_brueckner_transaction_list()
+                .fintech_calls_consent_activation_for_current_authorization_id()
+                .open_banking_can_read_anton_brueckner_transactions_data_using_consent_bound_to_service_session(
+                        ANTON_BRUECKNER_RESOURCE_ID, DATE_FROM, DATE_TO, BOTH_BOOKING
+                );
+    }
+
+    @Sql(statements = "UPDATE opb_bank_profile SET try_to_use_preferred_approach = true, preferred_approach = 'EMBEDDED' WHERE bank_uuid = '53c47f54-b9a4-465a-8f77-bc6cd5f0cf46'")
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java:198]: Please use JPA repository for this

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java (hunk 6 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java (hunk 7 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java (hunk 6 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java (hunk 8 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java (hunk 10 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java (hunk 13 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java (hunk 14 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java (hunk 16 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java (hunk 17 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java (hunk 15 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockConsentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java (hunk 9 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java (hunk 11 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java (hunk 12 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAuthenticatedPaymentE2EXs2aProtocolTest.java]: Done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java (hunk 7 above)
Comment (ivaniuminov) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockAnonymousPaymentE2EXs2aProtocolTest.java]: Done

**Pull request merged**